pkg/storaged/dialog.jsx

```diff
@@ -386,36 +415,19 @@ export const dialog_open = (def) => {
                     caption: def.Action.Title,
                     style: (def.Action.Danger || def.Action.DangerButton) ? "danger" : "primary",
                     disabled: running_promise != null,
-                    clicked: function (progress_callback) {
-                        const func = () => {
-                            return validate()
-                                    .then(() => {
-                                        const visible_values = { };
-                                        fields.forEach(f => {
-                                            if (is_visible(f, values))
-                                                visible_values[f.tag] = values[f.tag];
-                                        });
-                                        if (def.Action.wrapper)
-                                            return def.Action.wrapper(visible_values, progress_callback,
-                                                                      def.Action.action);
-                                        else
-                                            return def.Action.action(visible_values, progress_callback);
-                                    })
-                                    .catch(errs => {
-                                        if (errs && errs.toString() != "[object Object]") {
-                                            // Log errors from failed actions, for debugging and
-                                            // to allow the test suite to catch known issues.
-                                            console.warn(errs.toString());
-                                        }
-                                        errors = errs;
-                                        update();
-                                        return Promise.reject();
-                                    });
-                        };
-                        return client.run(func);
-                    }
+                    clicked: progress_callback => run_action(progress_callback, null),
                 }
             ];
+
+            if (def.Action.Variants)
+                for (const v of def.Action.Variants) {
+                    actions.push({
+                        caption: v.Title,
+                        style: (def.Action.Danger || def.Action.DangerButton) ? "danger" : "secondary",
+                        disabled: running_promise != null,
+                        clicked: progress_callback => run_action(progress_callback, v.tag),
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17888-20221115-130151-a219e579-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)